src/main/webapp/assets/common/js/gitbucket.js

```diff
@@ -763,3 +763,344 @@ var applyTaskListCheckedStatus = function(commentArea, checkboxes) {
   ss.pop();
   return ss.join('');
 };
+
+/**
+ * helper function for markdown toolbar operation
+ * check if index position is the middle of a word.
+ * @param {String} txt
+ * @param {Number} pos position in 'txt'
+ * @returns {Boolean}
+ */
+function isInWord(txt, pos){
+  if(pos <= 0){
+    return false;
+  }else if(pos === txt.length){
+    return false;
+  }else{
+    return (txt[pos - 1].match(/\s/g) === null) && (txt[pos].match(/\s/g) === null);
+  }
+}
+
+/**
+ * helper function for markdown toolbar operation
+ * get index of start position and end position according to pattern.
+ * if pattern is '\s' means word, if '\n' means line.
+ * @param {String} txt
+ * @param {Number} pos position in 'txt'
+ * @param {String} pattern e.g. ' ', '\n'
+ * @returns Array[Number, Number]
```

Comment:
Ah, JSDoc may not support tuple yet...

Reply:
@SIkebe Thank you for your review! I fixed JSDoc at https://github.com/gitbucket/gitbucket/pull/2528/commits/3611da3c94f9134346a50fb3a2150cbcb089289e.